src/ansys/fluent/core/launcher/process_launch_string.py

```diff
@@ -34,13 +36,26 @@
     Precision,
     UIMode,
 )
+from ansys.fluent.core.pyfluent_warnings import PyFluentUserWarning
 from ansys.fluent.core.scheduler import build_parallel_options, load_machines
 from ansys.fluent.core.utils.fluent_version import FluentVersion
 
 _THIS_DIR = os.path.dirname(__file__)
 _OPTIONS_FILE = os.path.join(_THIS_DIR, "fluent_launcher_options.json")
 
 
+def _get_default_ui_mode(ui_mode: UIMode | None = None) -> UIMode:
+    """Get the default UI mode based on the operating system."""
+    if platform.system() == "Linux":
+        return UIMode.NO_GUI
+    else:
+        return ui_mode
```

Comment:
@prmukherj Please could you explain the logic here? Is this correct? This seems to be overriding the mode that has been input by the user, thereby completely ignoring the user's selection. What am I missing?

Reply:
It is for getting the default mode for a specific system, right? If the system is linux the default mode is no_gui. Is it not true?

Reply:
Although we already have a default behaviour that is platform dependent already:
![image](https://github.com/user-attachments/assets/54f86fd4-bc75-4d97-93ef-a2e4e13bcd6b)

Reply:
We don't need `_get_default_ui_mode` it is alreday handled by `_default`.